client/lib/signup/step-actions/index.js

```diff
@@ -174,6 +174,24 @@ function getNewSiteParams( {
 	// when segment and vertical values are not sent. Check pbAok1-p2#comment-834.
 	const shouldUseDefaultAnnotationAsFallback = true;
 
+	// If there's a selected design, use similar new site options to what used
+	// in gutenboarding (see `client/landing/gutenboarding/stores/onboard/actions.ts`)
+	const selectedDesign = get( signupDependencies, 'selectedDesign', false );
+	const withDesignOptions = {
+		theme: `pub/${ selectedDesign?.theme || themeSlugWithRepo }`,
```

Comment:
Not entirely sure if, in case `selectedDesign.?theme` is falsey, we should prepend `pub/` also to `themeSlugWithRepo`.

Also, is there a reason why we wouldn't use `theme` instead of `themeSlugWithRepo` ?

Reply:
Probably we need to rethink the creation of this object since [we use it](https://github.com/Automattic/wp-calypso/pull/51501/files#diff-a621c4ea73ecba2ef5f903f325105824ebd61dcc12da5f505d2e0031bc4a4fc9R214) only when `selectedDesign` is defined and, in that case, it always includes `theme`, `template` and `fonts` properties.

Reply:
I partially reworked this logic in https://github.com/Automattic/wp-calypso/pull/51501/commits/70f217406b700d0db5e962d19f2bc95d466aab74 and https://github.com/Automattic/wp-calypso/pull/51501/commits/09c123090c56ccf20c99a60aba96eee2efa693dd

Reply:
Looks good! Thanks!